lte/gateway/python/magma/mobilityd/main.py

```diff
@@ -10,23 +10,109 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
+import logging
 
+from magma.common.redis.client import get_default_client
 from magma.common.service import MagmaService
+from magma.common.service_registry import ServiceRegistry
+from magma.mobilityd.ip_address_man import IPAddressManager
+from magma.mobilityd.ip_allocator_dhcp import IPAllocatorDHCP
+from magma.mobilityd.ip_allocator_pool import IpAllocatorPool
+from magma.mobilityd.ip_allocator_static import IPAllocatorStaticWrapper
+from magma.mobilityd.ip_allocator_multi_apn import IPAllocatorMultiAPNWrapper
+from magma.mobilityd.ipv6_allocator_pool import IPv6AllocatorPool
 from magma.mobilityd.rpc_servicer import MobilityServiceRpcServicer
+from magma.mobilityd.mobility_store import MobilityStore
 from lte.protos.mconfig import mconfigs_pb2
-from magma.common.service_registry import ServiceRegistry
 from lte.protos.subscriberdb_pb2_grpc import SubscriberDBStub
 
 
+def _get_ipv4_allocator(store: MobilityStore, allocator_type: int,
+                        static_ip_enabled: bool, multi_apn: bool,
+                        dhcp_iface: str, dhcp_retry_limit: int,
+                        subscriberdb_rpc_stub: SubscriberDBStub = None):
+    if allocator_type == mconfigs_pb2.MobilityD.IP_POOL:
+        store.dhcp_gw_info.read_default_gw()
+        ip_allocator = IpAllocatorPool(store.assigned_ip_blocks,
+                                       store.ip_state_map,
+                                       store.sid_ips_map)
+    elif allocator_type == mconfigs_pb2.MobilityD.DHCP:
+        ip_allocator = IPAllocatorDHCP(
+            assigned_ip_blocks=store.assigned_ip_blocks,
+            ip_state_map=store.ip_state_map,
+            iface=dhcp_iface,
+            retry_limit=dhcp_retry_limit,
+            dhcp_store=store.dhcp_store,
+            gw_info=store.dhcp_gw_info)
+    else:
+        raise ValueError(
+            "Unknown IP allocator type: %s" % allocator_type)
+
+    if static_ip_enabled:
+        ip_allocator = IPAllocatorStaticWrapper(
+            subscriberdb_rpc_stub=subscriberdb_rpc_stub,
+            ip_allocator=ip_allocator,
+            gw_info=store.dhcp_gw_info,
+            assigned_ip_blocks=store.assigned_ip_blocks,
+            ip_state_map=store.ip_state_map)
```

Comment:
inject just the `store` into the constructors of these components, and track the store as an instance member instead of the composed members. this will give us a clean way forward to turning that component into a general purpose DAO for mobilityd